src/chainstate/burn/db/sortdb.rs

```diff
@@ -2430,6 +2463,148 @@ impl SortitionDB {
         Ok(())
     }
 
+    fn apply_schema_3(tx: &mut SortitionHandleTx) -> Result<(), db_error> {
+        // fill in block_commit_parents table
+        info!("Populating block-commit parent table (this might take a while)");
+
+        let tip_sn = SortitionDB::get_canonical_burn_chain_tip(tx.deref())?;
+        let max_height = tip_sn.block_height;
+        let min_height = SortitionDB::get_first_block_snapshot(tx.deref())?.block_height;
+
+        for height in min_height..(max_height + 1) {
+            // load up the block-commits in this sortition history at this height
+            let ancestor_sn = match tx.get_block_snapshot_by_height(height)? {
+                Some(sn) => sn,
+                None => {
+                    if height == tip_sn.block_height {
+                        tip_sn.clone()
+                    } else {
+                        panic!(
+                            "FATAL: missing sortition in burnchain database at height {}",
+                            height
+                        );
+                    }
+                }
+            };
+            let commits: Vec<LeaderBlockCommitOp> = {
+                let qry = "SELECT * FROM block_commits WHERE block_height = ?1 AND sortition_id = ?2 ORDER BY vtxindex ASC";
+                let args: &[&dyn ToSql] = &[
+                    &u64_to_sql(ancestor_sn.block_height)?,
+                    &ancestor_sn.sortition_id,
+                ];
+                let mut stmt = tx.prepare(qry)?;
+                let query_values = stmt
+                    .query_and_then(args, |row| {
+                        let commit = LeaderBlockCommitOp::from_row(&row)
+                            .expect("FATAL: failed to load block-commit");
+                        let res: Result<_, rusqlite::Error> = Ok(commit);
+                        res
+                    })
+                    .expect("FATAL: failed to query block_commits table");
+                // .collect() is not implemented for AndThenRows
+                let mut ret = vec![];
+                for value in query_values {
+                    ret.push(value?);
+                }
+                ret
```

Comment:
Couldn't you use `util_lib::db::query_rows` here?

Reply:
Fixed

Reply:
Maybe you missed a `git push`? I don't see any changes yet in the PR.

Reply:
Weird. Fixed.

Reply:
Same goes for the other comment threads -- I'm not seeing any of the changes from those either.

Reply:
How bizarre.  I remember pushing them.  Lemme do them again.